canvas/animate.go

```diff
@@ -0,0 +1,53 @@
+package canvas
+
+import (
+	"image/color"
+	"time"
+
+	"fyne.io/fyne"
+)
+
+const (
+	// DurationStandard is the time a standard interface animation will run.
+	DurationStandard = time.Millisecond * 300
+)
+
+func NewColorAnimator(start, stop color.Color, d time.Duration, fn func(color.Color)) *fyne.Animation {
```

Comment:
This way to expose the animation API is hard to scale, more properties more different methods to use.

Maybe it can be abstracted with some mapping using consts

i.e.:

```
type Property int
const (
  Unknown = iota
  Color
  X
  Y
  Width
  Height
 // Newer properties 
)

// A Property to value mapping table near the defined values

func NewAnimator(start, stop float64, prop Property, d time.Duration, fn func(v float64)) *fyne.Animation {
```

the logic of the function will make the mapping to the corresponding attribute and make it change over the time

Reply:
This sort of API should be added as well, though this relates to the object-specific animation so you need to pass the `CanvasObject` as well. In this situation it may require that the animation is tied to the canvas, so likely separate / additional to this API which allows more generic / flexible API.

Reply:
Just for clarity these methods will not multiply out with each property, there are only multiple methods right now for the different types to avoid animations having to do the calcuations of "completion" -> "colour value" for example.

Reply:
It is also possible that property-based animators might relate to the DataBinding API also landing in 2.0.

Reply:
If you're happy can you mark this conversation resolved please @markcial?